src/server/rpc/procedures/baidu-maps/baidu-maps.js

```diff
@@ -0,0 +1,357 @@
+ * @param {Number} x x position of the point
+ * @param {Number} y y position of the point
+ * @returns {Array} A list containing the latitude and longitude of the given point.
+ */
+
+BaiduMaps.prototype.getEarthCoordinates = function(x, y){
+    return this._getUserMap(this.caller.clientId).then(mapInfo => {
+        let coords = this._coordsAt(x,y, mapInfo);
+        return [coords.lat, coords.lon];
+    });
+};
+
+/**
+ * Get the image coordinates (x, y) of a given location on the earth (latitude, longitude).
+ * @param {Latitude} latitude latitude of the point
+ * @param {Longitude} longitude longitude of the point
+ * @returns {Array} A list containing (x, y) position of the given point.
+ */
+
+BaiduMaps.prototype.getImageCoordinates = function(latitude, longitude){
+    return this._getUserMap(this.caller.clientId).then(mapInfo => {
+        let pixels = this._pixelsAt(latitude, longitude, mapInfo);
+        return [pixels.x, pixels.y];
+    });
+};
+
+/**
+ * Get the straight line distance between two points in meters.
+ * @param {Latitude} startLatitude Latitude of start point
+ * @param {Longitude} startLongitude Longitude of start point
+ * @param {Latitude} endLatitude Latitude of end point
+ * @param {Longitude} endLongitude Longitude of end point
+ * @returns {Number} Distance in meters
+ */
+BaiduMaps.prototype.getDistance = function(startLatitude, startLongitude, endLatitude, endLongitude){
+    return geolib.getDistance(
+        {latitude: startLatitude, longitude: startLongitude},
+        {latitude: endLatitude, longitude: endLongitude}
+    );
+};
+
+// Getting current map settings
+BaiduMaps.prototype._getUserMap = function() {
+    return this._getMapInfo(this.caller.clientId).then(map => {
+        if (!map) {
+            throw new Error('No map found. Please request a map and try again.');
+        }
+        return map;
+    });
+};
+
+var mapGetter = function(minMax, attr) {
+    return function() {
+        return this._getUserMap(this.caller.clientId)
+            .then(map => map[minMax][attr]);
+    };
+};
+
+/**
+ * Get the maximum longitude of the current map.
+ * @returns {Longitude}
+ */
+BaiduMaps.prototype.maxLongitude = mapGetter('max', 'lon');
+
+/**
+ * Get the maximum latitude of the current map.
+ * @returns {Longitude}
+ */
+BaiduMaps.prototype.maxLatitude = mapGetter('max', 'lat');
+
+/**
+ * Get the minimum longitude of the current map.
+ * @returns {Longitude}
+ */
+BaiduMaps.prototype.minLongitude = mapGetter('min', 'lon');
+
+/**
+ * Get the minimum latitude of the current map.
+ * @returns {Longitude}
+ */
+BaiduMaps.prototype.minLatitude = mapGetter('min', 'lat');
+
+BaiduMaps.isSupported = () => {
+    if(!key){
+        /* eslint-disable no-console*/
+        console.error('BAIDU_MAPS_KEY is missing.');
+        /* eslint-enable no-console*/
+    }
+    return !!key;
+};
+// Map of argument name to old field name
+BaiduMaps.COMPATIBILITY = {
```

Comment:
None of the compatibility options are needed. In the GoogleMaps service, these are only for backwards-compatibility.

Reply:
Are they used in the existing Google Maps examples?

Reply:
I am not sure but they are likely used within other user projects in the database.